mmda/models/base_document_predictor.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from abc import abstractmethod
+from typing import Union, List, Dict, Any
+
+from mmda.types.annotation import Annotation
+from mmda.types.document import Document
+
+
+class BaseDocumentPredictor:
+
+    ###################################################################
+    ##################### Necessary Model Variables ###################
+    ###################################################################
+
+    # TODO[Shannon] Add the check for required backends in the future.
+    # So different models might require different backends:
+    # For example, LayoutLM only needs transformers, but LayoutLMv2
+    # needs transformers and Detectron2. It is the model creators'
+    # responsibility to check the required backends.
+    @property
+    @abstractmethod
+    def REQUIRED_BACKENDS(self):
+        return None
+
+    @property
+    @abstractmethod
+    def REQUIRED_DOCUMENT_FIELDS(self):
```

Comment:
i'd remove this.  it's insufficient to make things tied to fieldnames that are agnostic to the model that produced that field.  for example, Model A produces `sents` might be different from Model B's `sents`.  if your Model C depends on Model A's `sents` and not Model B's `sents`, then just storing `sents` isnt sufficient.

for now, let's not worry about enforcing them in the implementation.  start w/ good documentation, release practices

Reply:
Yup -- but that might cause some bugs? Because we need to call `doc.sents` and it's probably reasonable to check that beforehand?